### ISSUE TITLE
URL#openStream returns null instead of throwing IOException

### ISSUE THREAD

**Issue by kristofdho** [oracle/graal#3164]
**Describe the issue**
The substituted implementation of URL@openStream for internal resources, returns null for non existing resources instead of throwing an IOException, as java does.
The provided example reflects how javafx searches for available upscaled resources for hidpi scaling.

**Steps to reproduce the issue**
Please include both build steps as well as run steps
1. **source file: URLConnectionPoc.java**
```java
import java.net.URL;

class URLConnectionPoc {
    public static void main(String[] args) {
        String file = "file.txt";
        try {
            URL url = URLConnectionPoc.class.getClassLoader().getResource(file);
            System.out.println("url: " + url);
            String newUrlStr = modifyUrl(url.toString());
            System.out.println("newUrl: " + newUrlStr);
            URL newUrl = new URL(newUrlStr);
            System.out.println("stream: " + newUrl.openStream());
        } catch (Throwable t) {
            t.printStackTrace();
        }
    }

    public static String modifyUrl(String path) {
        StringBuilder result = new StringBuilder();
        int dot = path.lastIndexOf(".");
        result.append(path, 0, dot);
        result.append("@2x");
        result.append(path.substring(dot));
        return result.toString();
    }
}
```

2. **create a new file "file.txt" in the same directory as URLConnectionPoc.java**
3. **compile source code and native image and create a jar**
```cmd
javac URLConnectionPoc.java
jar cvf URLConnectionPoc.jar URLConnectionPoc.class file.txt
native-image -H:Name=URLConnectionPoc -H:Class=URLConnectionPoc -H:IncludeResources=file.txt --verbose --no-fallback
```
4. **run java version for reference**
```cmd
C:\url-connection-poc>java URLConnectionPoc
url: file:/C:/url-connection-poc/file.txt
newUrl: file:/C:/url-connection-poc/file@2x.txt
java.io.FileNotFoundException: C:\url-connection-poc\file@2x.txt (The system cannot find the file specified)
        at java.base/java.io.FileInputStream.open0(Native Method)
        at java.base/java.io.FileInputStream.open(FileInputStream.java:219)
        at java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)
        at java.base/java.io.FileInputStream.<init>(FileInputStream.java:112)
        at java.base/sun.net.www.protocol.file.FileURLConnection.connect(FileURLConnection.java:86)
        at java.base/sun.net.www.protocol.file.FileURLConnection.getInputStream(FileURLConnection.java:184)
        at java.base/java.net.URL.openStream(URL.java:1140)
        at URLConnectionPoc.main(URLConnectionPoc.java:12)
```
5. **run the jar for reference**
```cmd
C:\url-connection-poc>java -cp URLConnectionPoc.jar URLConnectionPoc
url: jar:file:/C:/url-connection-poc/URLConnectionPoc.jar!/file.txt
newUrl: jar:file:/C:/url-connection-poc/URLConnectionPoc.jar!/file@2x.txt
java.io.FileNotFoundException: JAR entry file@2x.txt not found in C:\url-connection-poc\URLConnectionPoc.jar
        at java.base/sun.net.www.protocol.jar.JarURLConnection.connect(JarURLConnection.java:147)
        at java.base/sun.net.www.protocol.jar.JarURLConnection.getInputStream(JarURLConnection.java:155)
        at java.base/java.net.URL.openStream(URL.java:1140)
        at URLConnectionPoc.main(URLConnectionPoc.java:12)
```

6. **run native-image**
```cmd
C:\url-connection-poc>URLConnectionPoc.exe
url: resource:file.txt
newUrl: resource:file@2x.txt
stream: null
```

The issue is located here:
https://github.com/oracle/graal/blob/cf4cf49c7858b367cadfe08b534eccfd08d6b197/substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/JavaNetSubstitutions.java#L194-L204

A nullpointer is returned while an IOException should be thrown to behave the same as in java.

**Describe GraalVM and your environment:**
 - GraalVM version: **CE 21.0.0**
 - JDK major version: **11**
 - OS: **Windows 10**
 - Architecture: **x64**

**Comment by mcraj017** [comment]
@kkriske Thanks for the PR
We will check it out and move forward

**Comment by teshull** [comment]
Fixed: 708144f42925b7116703b70a3824fe6676523c5d
